diff --git a/node_modules/react-native-image-crop-picker/ios/src/ImageCropPicker.m b/node_modules/react-native-image-crop-picker/ios/src/ImageCropPicker.m
index 7101410..0f370fc 100644
--- a/node_modules/react-native-image-crop-picker/ios/src/ImageCropPicker.m
+++ b/node_modules/react-native-image-crop-picker/ios/src/ImageCropPicker.m
@@ -531,13 +531,14 @@ - (void)qb_imagePickerController:
     options.networkAccessAllowed = YES;
     
     if ([[[self options] objectForKey:@"multiple"] boolValue]) {
-        NSMutableArray *selections = [[NSMutableArray alloc] init];
+        NSMutableArray *sortedSelections = [[NSMutableArray alloc] init];
         
         [self showActivityIndicator:^(UIActivityIndicatorView *indicatorView, UIView *overlayView) {
             NSLock *lock = [[NSLock alloc] init];
             __block int processed = 0;
             
-            for (PHAsset *phAsset in assets) {
+            for (int i = 0; i < assets.count; i++) {
+                PHAsset *phAsset = assets[i];
                 
                 if (phAsset.mediaType == PHAssetMediaTypeVideo) {
                     [self getVideoAsset:phAsset completion:^(NSDictionary* video) {
@@ -553,7 +554,10 @@ - (void)qb_imagePickerController:
                                 return;
                             }
                             
-                            [selections addObject:video];
+                            [sortedSelections addObject:@{
+                                @"order": [NSString stringWithFormat:@"%5d", i],
+                                @"data": video
+                            }];
                             processed++;
                             [lock unlock];
                             
@@ -561,6 +565,15 @@ - (void)qb_imagePickerController:
                                 [indicatorView stopAnimating];
                                 [overlayView removeFromSuperview];
                                 [imagePickerController dismissViewControllerAnimated:YES completion:[self waitAnimationEnd:^{
+                                    [sortedSelections sortUsingComparator:^NSComparisonResult(id  _Nonnull obj1, id  _Nonnull obj2) {
+                                        return [[obj1 valueForKey:@"order"] compare:[obj2 valueForKey:@"order"]];
+                                    }];
+                                    
+                                    NSMutableArray *selections = [[NSMutableArray alloc] init];
+                                    for (NSDictionary *dict in sortedSelections) {
+                                        [selections addObject:dict[@"data"]];
+                                    }
+                                    
                                     self.resolve(selections);
                                 }]];
                                 return;
@@ -627,7 +640,7 @@ - (void)qb_imagePickerController:
                                         exif = [[CIImage imageWithData:imageData] properties];
                                     }
                                     
-                                    [selections addObject:[self createAttachmentResponse:filePath
+                                    NSDictionary *data = [self createAttachmentResponse:filePath
                                                                                 withExif: exif
                                                                            withSourceURL:[sourceURL absoluteString]
                                                                      withLocalIdentifier: phAsset.localIdentifier
@@ -641,7 +654,11 @@ - (void)qb_imagePickerController:
                                                                                 withRect:CGRectNull
                                                                         withCreationDate:phAsset.creationDate
                                                                     withModificationDate:phAsset.modificationDate
-                                                           ]];
+                                                           ];
+                                    [sortedSelections addObject:@{
+                                        @"order": [NSString stringWithFormat:@"%5d", i],
+                                        @"data": data
+                                    }];
                                 }
                                 processed++;
                                 [lock unlock];
@@ -651,6 +668,14 @@ - (void)qb_imagePickerController:
                                     [indicatorView stopAnimating];
                                     [overlayView removeFromSuperview];
                                     [imagePickerController dismissViewControllerAnimated:YES completion:[self waitAnimationEnd:^{
+                                        [sortedSelections sortUsingComparator:^NSComparisonResult(id  _Nonnull obj1, id  _Nonnull obj2) {
+                                            return [[obj1 valueForKey:@"order"] compare:[obj2 valueForKey:@"order"]];
+                                        }];
+                                        
+                                        NSMutableArray *selections = [[NSMutableArray alloc] init];
+                                        for (NSDictionary *dict in sortedSelections) {
+                                            [selections addObject:dict[@"data"]];
+                                        }
                                         self.resolve(selections);
                                     }]];
                                     return;
